modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -394,6 +396,8 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
           line"def $apply_[I, E, O, SI, SO](fa: service.Endpoint[I, E, O, SI, SO]): P2[I, E, O, SI, SO] =",
           if (ops.isEmpty) line"""sys.error("impossible")"""
           else
+            // Should be:
+            // line"fa.runWithProduct(algebra)"
```

Comment:
this is where compilation would fail because it sees `fa` with type `smithy4s.Endpoint[...`